### PR TITLE
Fix several crashes in DPL/Lightweight solution load.

### PR DESCRIPTION
**Pull request by Pilchie**
Internal bugs 269571, 274711, 272652, 273134, and 268059.

### CODE REVIEW

**Comment by Pilchie** [comment]
Tagging @MattGertz and @dotnet/roslyn-ide for review.

@dotnet-bot test vsi please

**Comment by tannergooding** [comment]
test vsi please

File: src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker_IVsSolutionLoadEvents.cs [hunk 1]
```diff
@@ -218,13 +218,23 @@ private void OutputToOutputWindow(string message)
             OutputToOutputWindow($"\tCreating '{projectName}':\t{commandLineArguments.SourceFiles.Length} source files,\t{commandLineArguments.MetadataReferences.Length} references.");
```
Comment (rchande) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker_IVsSolutionLoadEvents.cs]: Unrelated but does this need to be loc'd?

Comment (Pilchie) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker_IVsSolutionLoadEvents.cs]: No - this is [Conditional("DEBUG")] - it never shows up in customer builds, it's just convenient for debugging.

**Comment by rchande** [comment]
👍

**Comment by Pilchie** [comment]
@dotnet-bot test vsi please again, now that Jenkins knows how to run vsi tests against `dev15-rc`.

**Comment by MattGertz** [comment]
Approved pending passing tests.

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by Pilchie** [comment]
@dotnet-bot retest windows_release_unit32_prtest please - hit the dependency copy race.

**Comment by Pilchie** [comment]
@dotnet-bot test vsi please

**Comment by Pilchie** [comment]
@mmitche - I can't seem to get the vsi tests to notice that they failed, or the failed test to re-run.  Any ideas?

**Comment by tannergooding** [comment]
@Pilchie, it looks like the only one remaining is still actively running. The others appear to have actually passed.

**Comment by Pilchie** [comment]
Nice - my retest must have (eventually) worked.

**Comment by Pilchie** [comment]
@dotnet-bot retest windows_release_unit32_prtest please - hit the dependency build race issue.

**Comment by Pilchie** [comment]
@dotnet-bot retest windows_release_unit32_prtest please - hit the dependency build race issue.

**Comment by Pilchie** [comment]
@dotnet-bot test windows_release_unit32_prtest please

Hit the dependency build race issue.

**Comment by Pilchie** [comment]
@dotnet-bot test windows_release_unit32_prtest please

Hit the dependency build race issue.

**Comment by Pilchie** [comment]
@dotnet-bot retest windows_release_unit32_prtest please

Hit the dependency build race issue.

**Comment by CyrusNajmabadi** [comment]
retest windows_release_unit32_prtest please

**Pull request merged**